kyo-core/shared/src/main/scala/kyo/env.scala

```diff
@@ -0,0 +1,47 @@
+package kyo
+
+import kyo.Tag.Intersection
+import scala.collection.immutable.HashMap
+
+opaque type Env[+R] = HashMap[Tag[?], Any]
+extension [R](self: Env[R])
+
+    private def fatal(using t: Tag[?]): Nothing =
+        throw new RuntimeException(s"fatal: kyo.Env of contents [$self] missing value of type: [${t.show}].")
+
+    def get[A >: R](using Tag[A]): A =
+        getOrElse(fatal)
+
+    def getOrElse[A >: R](default: => A)(using t: Tag[A]): A =
+        val result = self.getOrElse(t, null)
```

Comment:
Could `default` be passed to `getOrElse` directly?

Reply:
A is basically guaranteed to exist in the map, as long as we expose the right interface (and people don't unsafely cast). So I don't think we need getOrElse?

Reply:
I think we can remove it. Realistically, this method is only useful for internal usage. Later we can re-expose as package private.